packages/amplify-provider-awscloudformation/src/push-resources.js

```diff
@@ -494,6 +611,18 @@ function formNestedStack(context, projectDetails, categoryName, resourceName, se
   const nestedStack = context.amplify.readJsonFile(initTemplateFilePath);
   const { amplifyMeta } = projectDetails;
   let authResourceName;
+
+  const { VpcCfnFile } = amplifyMeta.providers[constants.ProviderName];
```

Comment:
Why can't we keep the network CFN file all the time and use cfn conditionl to deploy it or not. And set the condition only when it is needed. This way we don't need to push it separately and there isn't any magic in terms of the CFN templates. The current approach customer can't introspect the template easily